date.pc.in

```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_BINDIR@
+libdir=@CMAKE_INSTALL_LIB@
+includedir=@CMAKE_INSTALL_INCLUDE@
```

Comment:
We can go one step further... this should be using https://github.com/jtojnar/cmake-snips/#concatenating-paths-when-building-pkg-config-files

Reply:
Personally that seems like overkill, there are a lot of packages in nixpkgs that just use the CMAKE_INSTALL_* variables in their pkgconfig files, but I might be missing some use cases where the concatenation is necessary

Reply:
I would use the `CMAKE_INSTALL_FULL_*` variables for `includedir` and `libdir`. Using e.g. `CMAKE_INSTALL_LIBDIR` is an open invitation to do things like `${prefix}/@CMAKE_INSTALL_LIBDIR@` or `@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@`, which explode in NixOS all the time (libdir and includedir can be disjoint paths) and whenever `CMAKE_INSTALL_LIBDIR` is set to an absolute path (e.g. NixOS). The CMake documentation is quite clear on the fact that one should not assume on `CMAKE_INSTALL_*` variables to be absolute or relative. If one needs a normalized version, then use `CMAKE_INSTALL_FULL_*`.
I would also remove `prefix` and `exec_prefix`, since they are not used in the pkg-config file and there may not actually be a common prefix for the paths. 

Sure, you loose the possibility to have a relocatable pkg-config file, but if you can build and install the project (including the generated pkg-config file) into another prefix, then you can also set the `PKG_CONFIG_PATH` accordingly and load the right pkg-config file.

Reply:
@panicgh 

This is literally the exact reason why @jtojnar (a NixOS developer) wrote the cmake-snips guide to doing this correctly.

It continually saddens me that despite the efforts of an actual, knowledgeable NixOS developer who both understands how build systems work, and cares about the ecosystem as a whole and wants to avoid breaking distros other than NixOS, and writes excellent documentation on how to do the right, spec-compliant thing... other NixOS people are always trying to demolish, destroy, and ruin @jtojnar's work, damage NixOS's reputation, and make software that doesn't work or occasionally break software that used to work.

I see this way too often. In some cases, after someone from NixOS went ahead and fixed things in accordance with the cmake-snips documentation, someone else from NixOS comes along and breaks it so badly that it stops working in NixOS either.

There is a correct way and a wrong way to do it. Writing NixOS specific files instead of files that work correctly both on NixOS and everywhere else, is the ***wrong*** way to do it.

My original recommendation is the correct way to do it. It would be great if NixOS users would stop trying to break the ecosystem through ignorance and then advocate that it's outright bad to use the correct way. Thanks.

(Why, why, why, WHY for the love of G-d is it always NixOS users??? 😞)